### PR TITLE
[eas-cli] Add device:list and device:view commands

### PR DESCRIPTION
**Pull request by satya164**
This adds commands to view registered devices for an account.

Screen Shot 2020-12-10 at 16 28 59

Screen Shot 2020-12-10 at 16 38 29

Depends on https://github.com/expo/universe/pull/6614

### CODE REVIEW

File: packages/eas-cli/src/commands/device/list.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+          if (teams.length === 1) {
+            appleTeamIdentifier = teams[0].appleTeamIdentifier;
+          } else {
+            const result = await promptAsync({
+              type: 'select',
+              name: 'appleTeamIdentifier',
+              message: 'What Apple Team would you like to list devices for?',
+              choices: teams.map(team => ({
+                title: team.appleTeamName ?? team.id,
```
Comment (satya164) [inline packages/eas-cli/src/commands/device/list.ts:45]: I'm not sure what to show in the prompt if there is no name

Comment (wkozyra95) [reply packages/eas-cli/src/commands/device/list.ts:45]: Maybe if there is team name display both values and if it's only id then only that one value.
- `teamId - Team Name`
- `teamID (Team Name)`

**Review by wkozyra95 — APPROVED** [review]

File: packages/eas-cli/src/commands/device/list.ts (hunk 1 above)
Comment (dsokal) [reply packages/eas-cli/src/commands/device/list.ts:45]: My suggestion: `${team.appleTeamName} (ID: ${team.id})`

File: packages/eas-cli/src/commands/device/list.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        } else {
+          spinner.fail(`Couldn't find any teams for the account ${accountName}`);
+        }
+      } catch (e) {
+        spinner.fail(`Something went wrong and we couldn't fetch the list of teams`);
+        throw e;
+      }
+    }
+
+    if (appleTeamIdentifier) {
```
Comment (dsokal) [inline packages/eas-cli/src/commands/device/list.ts:61]: Maybe `assert(appleTeamIdentifier)` instead?

File: packages/eas-cli/src/graphql/queries/AppleDeviceQuery.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+export type AppleDevicesByAccountNameQueryResult = AppleTeamQueryResult & {
+  appleDevices: AppleDeviceQueryResult[];
+};
+
+export type AppleDevicesByIdentifierQueryResult = AppleDeviceQueryResult & {
+  appleTeam: AppleTeamQueryResult;
+};
+
+const AppleDeviceQuery = {
```
Comment (dsokal) [inline packages/eas-cli/src/graphql/queries/AppleDeviceQuery.ts:21]: I think it'd be better to have a query like this:
```
{
  account {
    byName(accountName: "dsokal-staging") {
      appleTeams(appleTeamIdentifier: "QL76XYH73P") {
        id
        appleTeamName
        appleDevices {
          id
          identifier
          name
          deviceClass
          enabled
        }
      }
    }
  }
}
```

Comment (dsokal) [reply packages/eas-cli/src/graphql/queries/AppleDeviceQuery.ts:21]: This way you don't need the account id and the query you added in a new file - `packages/eas-cli/src/graphql/queries/AccountQuery.ts`

**Comment by fson** [comment]
Nicely formatted output – looks beautiful! 😍

File: packages/eas-cli/src/graphql/queries/AccountQuery.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import gql from 'graphql-tag';
```
Comment (dsokal) [inline packages/eas-cli/src/graphql/queries/AccountQuery.ts:1]: It seems we don't need that query, see the comment for AppleDeviceQuery.ts.

File: packages/eas-cli/src/graphql/queries/AppleTeamQuery.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import gql from 'graphql-tag';
+
+import { graphqlClient, withErrorHandlingAsync } from '../client';
+import { AppleTeam } from '../generated';
+
+type AppleTeamQueryResult = Pick<AppleTeam, 'id' | 'appleTeamIdentifier' | 'appleTeamName'>;
+
+const AppleTeamQuery = {
+  async allForAccountAsync(accountName: string): Promise<AppleTeamQueryResult[]> {
```
Comment (dsokal) [inline packages/eas-cli/src/graphql/queries/AppleTeamQuery.ts:9]: Maybe move this function to `packages/eas-cli/src/credentials/ios/api/graphql/queries/AppleTeamQuery.ts`

**Review by dsokal — CHANGES_REQUESTED** [review]

**Review by dsokal — APPROVED** [review]

**Pull request merged**